services/user-management/src/main/java/com/foodie/user/service/UserService.java

```diff
@@ -1,53 +1,71 @@
 package com.foodie.user.service;
 
-import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import com.foodie.FoodieBaseRepository;
+import com.foodie.FoodieBaseResponse;
+import com.foodie.FoodieBaseService;
 import com.foodie.user.contracts.ChangePasswordRequest;
-import com.foodie.user.model.Role;
 import com.foodie.user.model.User;
-import com.foodie.user.repositories.RoleRepository;
 import com.foodie.user.repositories.UserRepository;
+import com.foodie.user.utils.FoodieUserDetails;
 
 import java.security.Principal;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
-@RequiredArgsConstructor
-public class UserService {
+public class UserService implements FoodieBaseService<User, UUID> {
 
     private final UserRepository repository;
-    private final RoleRepository roleRepository;
     private final PasswordEncoder passwordEncoder;
 
-    public Optional<User> findByIdAndDeletedAtIsNull(UUID id) {
-        return repository.findByIdAndDeletedAtIsNull(id);
+    @Override
+    public FoodieBaseRepository<User, UUID> getRepository() {
+        return repository;
     }
 
-    public User validateAndGetUser(UUID userId) {
-        return findByIdAndDeletedAtIsNull(userId).orElse(null);
+    public User validateAndGetUser(Object connectedUser) {
+
+        User retrievedUser;
+        if (connectedUser instanceof Principal) {
+            retrievedUser = ((FoodieUserDetails) ((UsernamePasswordAuthenticationToken) connectedUser).getPrincipal()).getUser();
+        } else if (connectedUser instanceof UUID) {
+            retrievedUser =  findByIdAndDeletedAtIsNull((UUID) connectedUser).orElse(null);
+        } else {
+            throw new IllegalArgumentException("Invalid connectedUser type");
+        }
+
+        if (retrievedUser == null || !retrievedUser.getIsActive()) {
+            throw new RuntimeException("Requested user not found or not active");
```

Comment:
Not a good idea to throw `RuntimeException` use a more specific exception or create a custom exception class

Reply:
Will do so